couchbase/src/api/mod.rs

```diff
@@ -155,6 +156,58 @@ impl Cluster {
         receiver.await.unwrap()
     }
 
+    /// Executes an analytics query
+    ///
+    /// # Arguments
+    ///
+    /// * `statement` - the analyticss statement to execute
+    /// * `options` - allows to pass in custom options
+    ///
+    /// # Examples
+    ///
+    /// Run an analytics query with default options.
+    /// ```no_run
+    /// # let cluster = Cluster::connect("127.0.0.1", "username", "password");
+    /// let result = cluster.search_query::<QueryStringQuery>(
+    ///    String::from("test"),
+    ///    QueryStringQuery::new(String::from("swanky")),
+    ///    SearchOptions::default(),
+    ///);
+    /// ```
+    ///
+    /// This will return an async result, which can be consumed:
+    /// ```no_run
+    /// # let cluster = Cluster::connect("couchbase://127.0.0.1", "Administrator", "password");
+    /// match cluster.cluster.search_query::<QueryStringQuery>(
+    ///    String::from("test"),
+    ///    QueryStringQuery::new(String::from("swanky")),
+    ///    SearchOptions::default(),
+    ///).await {
+    ///     Ok(mut result) => {
+    ///         for row in result.rows::<serde_json::Value>().next().await {
+    ///             println!("Found Row {:?}", row);
+    ///         }
+    ///     },
+    ///     Err(e) => panic!("Query failed: {:?}", e),
+    /// }
+    /// ```
+    /// See the [SearchResult](struct.SearchResult.html) for more information on what and how it can be consumed.
+    pub async fn search_query<T: SearchQuery>(
+        &self,
+        index: String,
```

Comment:
similar to the other options, for convenience, make this take a generic S where S: Into<String> .. then inside the code do index.into() -- if someone passes in a &str it will be cloned into an owned one, if it is a String already it is a noop